src/typelib.erl

```diff
@@ -37,6 +37,8 @@ pp_type({type, _, bounded_fun, [FunType, []]}) ->
     %% Bounded fun with empty constraints gets printed with a trailing "when"
     %% when pretty-printed as a spec (next clause)
     pp_type(?assert_type(FunType, function_type()));
+pp_type({type, Anno, record, [Name|Fields]}) when length(Fields) > 0 ->
+    pp_type({type, Anno, record, [Name]});
```

Comment:
Regarding the commit message "erl_pp does not like record fields": have you tried replacing erl_pp with erl_prettypr?

Reply:
No but I think it might boil down from the fact that there is another notation possible. If I use 

```erlang
1> typelib:parse_type("#r{f :: integer()}").
{type,1,record,
      [{atom,1,r},
       {type,1,field_type,[{atom,1,f},{type,1,integer,[]}]}]}
```

I think what `erl_pp` would like is for me to specify only the fields that have been modified from their base values? Not sure, could test it.

Reply:
`erl_pp` doesn't care what the record definition is, so it doesn't matter if all the fields are added. It is purely a formatting of the parse tree. It must be something else...

With two fields refined (even without a record definition), it works in the current master:

```Erlang
3> T2 = {type, 1, record, [{atom, 1, r}, {type, 1, field_type, [{atom,1,f},{type,1,integer,[]}]},{type,1,field_type,[{atom,1,g},{type,1,atom,[]}]}]}.
{type,1,record,
      [{atom,1,r},
       {type,1,field_type,[{atom,1,f},{type,1,integer,[]}]},
       {type,1,field_type,[{atom,1,g},{type,1,atom,[]}]}]}
4> typelib:pp_type(T2).                                                                                                                              
"#r{f :: integer(), g :: atom()}"
```

Reply:
It has to be `field_type` here, not <del><code>typed_record_field</code></del> as in the record definition.

Reply:
The problem was that it was just extra arguments for the `record` type. Like

```erlang
{type, _, record, [{atom, _, name}, {type, _, integer, _}, {type, _, atom, _}]}
```

Reply:
OK, resolved then?

Reply:
If we keep it like so instead of using `field_type` then yes.

Reply:
Will be removed to use `field_type`